crates/matrix-sdk/src/room/timeline/event_handler.rs

```diff
@@ -407,18 +409,41 @@ impl<'a, 'i> TimelineEventHandler<'a, 'i> {
             sender: self.meta.sender.to_owned(),
             content,
             reactions,
-            origin_server_ts: self.flow.origin_server_ts(),
+            timestamp: self.flow.timestamp(),
             is_own: self.meta.is_own_event,
             encryption_info: self.meta.encryption_info.clone(),
             raw: self.flow.raw_event().cloned(),
         };
 
         let item = Arc::new(TimelineItem::Event(item));
         match &self.flow {
-            Flow::Local { .. } => {
+            Flow::Local { timestamp, .. } => {
+                // Check if the latest event has the same date as this event.
+                if let Some(latest_event) = self
+                    .timeline_items
+                    .iter()
+                    .rfind(|item| item.as_event().is_some())
+                    .and_then(|item| item.as_event())
+                {
+                    let old_ts = latest_event.timestamp();
+
+                    if let Some(day_divider_item) =
+                        maybe_create_day_divider_from_timestamps(old_ts, *timestamp)
+                    {
+                        self.timeline_items
+                            .push_cloned(Arc::new(TimelineItem::Virtual(day_divider_item)));
+                    }
+                } else {
+                    // If there is not event item, there is no day divider yet.
```

Comment:
```suggestion
                    // If there is no event item, there is no day divider yet.
```

Reply:
This means that the first timeline item is always going to be a day divider, right?

Reply:
Yes, I couldn't figure out a case right now where it wouldn't be true.